ui/src/components/auth/LoginContainer.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { useDispatch } from 'react-redux';
+import { signin } from '../../reducers/userSlice';
+import axios from 'axios';
+import { Box, Button, Checkbox, Link, Stack, TextField, Typography } from '@mui/material';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { LoginFormFields } from './helpers/form.types';
+import { formSpacing, formWidth } from './helpers/constants';
+
+// zod validation
+const loginValidationSchema = z.object({
+  email: z.string().email({ message: 'Please enter a valid email' }).trim(),
+  password: z
+    .string()
+    .trim()
+    .min(6, { message: 'Password must be at least 6 characters long' })
+    .max(100, { message: 'Password must be at most 100 characters long' }),
```

Comment:
crazy... can we limit at 24 characters lol or 32 at most

Reply:
ahhaha can, ill put 32